frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetail.tsx

```diff
@@ -338,14 +361,38 @@ export function ObjectDetailHeader({
       </div>
       <div className="flex align-center">
         {canEdit && (
-          <Button
-            className="mr1"
-            icon={isEditing ? "eye" : "pencil"}
-            onClick={onToggleEditingModeClick}
-            iconSize={20}
-            onlyIcon
-            borderless
-          />
+          <>
+            {deleteRow ? (
+              <ModalWithTrigger
+                ref={deleteRowModal}
+                triggerElement={
+                  <Button
+                    className="mr1"
+                    icon={"trash"}
+                    iconSize={20}
+                    onlyIcon
+                    borderless
+                  />
+                }
+              >
+                <ConfirmContent
+                  title={t`Delete row`}
+                  content={""}
+                  onClose={() => deleteRowModal.current.toggle()}
+                  onAction={() => deleteRow()}
+                />
+              </ModalWithTrigger>
+            ) : null}
+
+            <Button
+              className="mr1"
+              icon={isEditing ? "eye" : "pencil"}
+              onClick={onToggleEditingModeClick}
+              iconSize={20}
+              onlyIcon
+              borderless
+            />
+          </>
```

Comment:
Can we move the modal to its own file? There's ongoing work on this component, don't want our prototyping work cause too much noise for other folks

Reply:
Move to `metabase/writeback/components/ObjectDetails/ActionHeader`